app/services/publication-service.js

```diff
@@ -380,10 +380,8 @@ export default class PublicationService extends Service {
   async deletePiece(piece) {
     const file = await piece.file;
     const documentContainer = await piece.documentContainer;
-    await Promise.all([
-      piece.destroyRecord(),
-      file.destroyRecord(),
-      documentContainer.destroyRecord(),
-    ]);
+    await file.destroyRecord();
+    await documentContainer.destroyRecord();
+    await piece.destroyRecord();
```

Comment:
I assume `file` and `documentContainer` can still be executed in parallel since they don't have any direct connection to each other. But since you tested it with this flow and we're sure that one works, it's ok for me to merge the PR like this. It will only be a marginal gain.

Reply:
Small gains maybe, but I do assume that cancelling an upload of many documents is not going to be done often in day to day work.